File: system-test/java/src/main/java/com/optivem/shop/dsl/port/then/steps/ThenCountry.java
package com.optivem.shop.dsl.port.then.steps;

import com.optivem.shop.dsl.port.then.steps.base.ThenStep;

public interface ThenCountry extends ThenStep<ThenCountry> {
    ThenCountry hasCountry(String country);

    ThenCountry hasTaxRate(double taxRate);

    ThenCountry hasTaxRateIsPositive();
}

File: system-test/typescript/src/testkit/dsl/port/then/steps/then-country.ts
import type { ThenStep } from './then-step.js';

export interface ThenCountry extends ThenStep<ThenCountry> {
  hasCountry(country: string): ThenCountry;
  hasTaxRate(taxRate: number): ThenCountry;
  hasTaxRateIsPositive(): ThenCountry;
}
